src/whip.js

```diff
@@ -582,13 +582,24 @@ class JanusWhipServer extends EventEmitter {
 				// Now that we have a response, trickle the candidates we received
 				if(candidates.length > 0 && this.janus)
 					await endpoint.handle.trickle(candidates);
-				// Read the ICE credentials and send them back
-				let sdpAnswer = result.jsep.sdp;
-				let serverUfrag = sdpAnswer.match(/a=ice-ufrag:(.*)\r\n/)[1];
-				let serverPwd = sdpAnswer.match(/a=ice-pwd:(.*)\r\n/)[1];
-				let payload =
-					'a=ice-ufrag:' + serverUfrag + '\r\n' +
-					'a=ice-pwd:' + serverPwd + '\r\n';
+				// Read the ICE credentials/candidates and send them back
+				let serverUfrag, serverPwd, serverCandidates = [];
+
+				let sdp = result.jsep.sdp
+				const sections = sdp.split(/\r?\nm=/);
+				if(sections.length > 2)
+					sdp = sections[0] + sections[1];
```

Comment:
I think this is broken, because it's omitting the `m=` that was used to split the SDP. As a result, the `m` line is never added to the fragment even if your code looks for it, since it's not there:

```
a=msid-semantic: WMS *audio 9 UDP/TLS/RTP/SAVPF 111
```

Changing it to something like this fixed it for me:

```
sdp = sections[0] + '\r\nm=' + sections[1];
```

Reply:
@youennf any chance you could address this comment? I was planning to merge this as I have some other changes planned, but can't do it as it is. Thanks!